source/common/ssl/context_impl.cc

```diff
@@ -262,9 +263,20 @@ bool ContextImpl::verifySubjectAltName(X509* cert, const std::string& subject_al
       if (altname->type == GEN_DNS) {
         ASN1_STRING* str = altname->d.dNSName;
         char* dns_name = reinterpret_cast<char*>(ASN1_STRING_data(str));
-        if (sanMatch(subject_alt_name, dns_name)) {
-          verified = true;
-          break;
+        for (auto& config_san : subject_alt_names) {
+          if (dNSNameMatch(config_san, dns_name)) {
+            verified = true;
+            break;
```

Comment:
@wattli yes, great catch. @myidpt can we make sure to have a test case that would have caught this bug.